beacon-chain/p2p/utils.go

```diff
@@ -60,6 +60,17 @@ func privKey(cfg *Config) (*ecdsa.PrivateKey, error) {
 		if err != nil {
 			return nil, err
 		}
+		if cfg.StaticPeerID {
+			rawbytes, err := priv.Raw()
+			if err != nil {
+				return nil, err
+			}
+			dst := make([]byte, hex.EncodedLen(len(rawbytes)))
+			hex.Encode(dst, rawbytes)
+			if err := file.WriteFile(defaultKeyPath, dst); err != nil {
+				return nil, err
+			}
+		}
 		return ecdsaprysm.ConvertFromInterfacePrivKey(priv)
```

Comment:
I think re-arranging the logical flow here with some early returns would make the precedence of the different key options more clear:
```
        // PrivateKey cli flag takes highest precedence.
        if cfg.PrivateKey != "" {
                return privKeyFromFile(cfg.PrivateKey)
        }
        defaultKeyPath := path.Join(cfg.DataDir, keyPath)
        _, err := os.Stat(defaultKeyPath)
        defaultKeysExist := !os.IsNotExist(err)
        if err != nil && defaultKeysExist {
                return nil, err
        }
        // Default keys have the next highest precendence, if they exist.
        if defaultKeysExist {
                return privKeyFromFile(defaultKeyPath)
        }
        // There are no keys on the filesystem, so we need to generate one.
        priv, _, err := crypto.GenerateSecp256k1Key(rand.Reader)
        if err != nil {
                return nil, err
        }
        // If the StaticPeerID flag is set, save the generated key as the default
        // key, so that it will be used by default on the next node start.
        if cfg.StaticPeerID {
                rawbytes, err := priv.Raw()
                if err != nil {
                        return nil, err
                }
                dst := make([]byte, hex.EncodedLen(len(rawbytes)))
                hex.Encode(dst, rawbytes)
                if err := file.WriteFile(defaultKeyPath, dst); err != nil {
                        return nil, err
                }
                // Read the key from the defaultKeyPath file just written
                // for the strongest guarantee that the next start will be the same as this one.
                return privKeyFromFile(defaultKeyPath)
        }
        return ecdsaprysm.ConvertFromInterfacePrivKey(priv)
```